src/Control/Lens/Fold.hs

```diff
@@ -361,6 +362,24 @@ filtered :: (Choice p, Applicative f) => (a -> Bool) -> Optic' p f a a
 filtered p = dimap (\x -> if p x then Right x else Left x) (either pure id) . right'
 {-# INLINE filtered #-}
 
+-- | Obtain a potentially empty 'IndexedFold' by taking the first element from another,
+-- potentially empty `Fold` and using it as an index.
+--
+-- The resulting optic can be composed with to filter another 'Lens', 'Iso', 'Getter', 'Fold' (or 'Traversal').
+--
+-- >>> [(Just 2, 3 :: Int), (Nothing, 4)] <&> filteredBy (_1 . _Just) <. _2 %@~ (*)
```

Comment:
Can we change to `& mapped . filteredBy ...`, that would be more lens-y. (at most one fancy operator per example: here it's definitely `%@~` :)